lib/resque_bus/server/views/bus.erb

```diff
@@ -9,34 +9,64 @@ if (agree)
 else
 	return false ;
 }
+
+function setSample() {
+  var text = document.getElementById("query_attributes").textContent;
+  var textArea = document.getElementById('querytext');
+  textArea.value = text;
+  return false;
+}
 // -->
 </script>
 
 <%
   app_hash = {}
   class_hash = {}
   event_hash = {}
+  query_string = params[:query].to_s.strip
+
+  query_attributes = nil
+  query_error = nil
+  if query_string.length > 0
+    begin
+      query_attributes = ::ResqueBus::Server::Helpers.parse_query(query_string)
+      raise "Not a JSON Object" unless query_attributes.is_a?(Hash)
```

Comment:
I think it would make the view lighter if we put this logic into the parser. This is already repeated with some of the logic in the `parse_query` fuction.